src/background/background-events.js

```diff
@@ -0,0 +1,28 @@
+import { REQUEST_PACKAGE_INFO_EVENT, RESPONSE_PACKAGE_INFO_EVENT } from '../events-shared';
+import advisories from './advisory/index';
+
+const listener = async ({ type, detail }, port) => {
+  if (type === REQUEST_PACKAGE_INFO_EVENT) {
+    const promises = await advisories(detail);
+    Object.entries(promises).forEach(([part, promise]) => {
+      promise.then((info) => {
+        port.postMessage({
+          type: RESPONSE_PACKAGE_INFO_EVENT,
+          detail: {
+            packageId: detail,
+            part,
+            info,
+          },
+        });
+      });
+    });
+  }
+
+  return true;
+};
+
+export const listen = () => {
+  chrome.runtime.onConnect.addListener((port) => {
+    port.onMessage.addListener(listener);
+  });
+};
```

Comment:
The code became a deep indentation, and also I'm struggling with the objects that passed between the events, each one requires a particular property for its use case.

I'm open to suggestions.